### PR TITLE
HIP-584: Implement gas_estimate integration tests for erc readonly

### LINKED ISSUE #5597: HIP-584: Provide integration tests for ERC read-only precompiles
### Problem

We need default implementations for ERC read-only precompiles, so we should copy needed logic from `hedera-services` into `hedera-mirror-web3`.

### Solution

Provide ERC precompile implementations for the following precompiles:

- [ ] #6433 
- [ ] #6434 
- [ ] #6435 
- [ ] #6436 
- [ ] #6437 
- [ ] #6438 
- [ ] #6439 
- [ ] #6440 
- [ ] #6441 
- [ ] #6442 

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by MrValioBg**
**Description**:

Adds integration tests for the ERC read-only precompiles and provides test cases for static and non-static contract functions.

* The new functions are added in the `ERCTestContract.sol ` following the same naming convention -- ending with `NonStatic`.
* In `ContractCallServiceERCTokenTest` we added tests for `estimate_gas` and modified both `estimate_gas` and `eth_call` to test both cases for the static and NonStatic contract functions.

**Related issue(s)**:

Fixes #5597 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceERCTokenTest.java [hunk 1]
```diff
@@ -20,32 +20,44 @@
...
+    @MethodSource("ercContractFunctionArgumentsProvider")
+    void ercReadOnlyPrecompileOperationsTest(final ErcContractReadOnlyFunctions ercFunction, final boolean isStatic) {
+        final var functionName = ercFunction.getName(isStatic);
         final var functionHash =
-                functionEncodeDecoder.functionHashFor(ercFunction.name, ERC_ABI_PATH, ercFunction.functionParameters);
+                functionEncodeDecoder.functionHashFor(functionName, ERC_ABI_PATH, ercFunction.functionParameters);
         final var serviceParameters = serviceParametersForExecution(functionHash, ERC_CONTRACT_ADDRESS, ETH_CALL, 0L);
+
+        if (ercFunction == ErcContractReadOnlyFunctions.TOKEN_URI) {
+            assertThat(contractCallService.processCall(serviceParameters)).isNotEqualTo(Address.ZERO.toString());
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceERCTokenTest.java:45]: Can't we compare the result of `tokenUri` to the NFT `metadata` which was persisted in DB? If there isn't metadata, we can add it in `ContractCallTestSetup`.

Comment (MrValioBg) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServiceERCTokenTest.java]: Done. Thanks :)

**Review by 0xivanov — APPROVED** [review]

**Review by stoyanov-st — APPROVED** [review]
LGTM

**Review by IvanKavaldzhiev — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**